
syntax = "proto3";

package test;

// Interface exported by the server.
service Hello 
{
    // Shutdown the server
    rpc Shutdown (ShutdownRequest) returns (ShutdownResponse) {}

    // Ping the server 
    rpc Ping (PingRequest) returns (PingResponse) {}

    // Server-side streaming
    rpc ServerStream (ServerStreamRequest) returns (stream ServerStreamResponse) {}

    // Client-side streaming
    rpc ClientStream (stream ClientStreamRequest) returns (ClientStreamResponse) {}
}

message ShutdownRequest
{
    string reason = 1;
}

message ShutdownResponse
{
    bool result = 1;
    string msg = 2;
}

message PingRequest
{
    // empty message
}

message PingResponse
{
    string msg = 1;
}

message ServerStreamRequest
{
    string msg = 1; // The string to be included in every response message
}

message ServerStreamResponse
{
    string msg = 1; 
    bool result = 2;
}

message ClientStreamRequest
{
    string msg = 1; // The string to be included in every response message
}

message ClientStreamResponse
{
    bool result = 1;
}

